### PR TITLE
Overmap changes

### PR DESCRIPTION
**Pull request by Kwask**
- Adds overmap controller
- Revamps how overmap is created
- Adds randomly generated asteroids

### CODE REVIEW

File: code/controllers/Processes/overmap.dm [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+/datum/controller/process/overmap/New()
+	..()
+
+	overmap = src
+
+/datum/controller/process/overmap/setup()
+	name = "Overmap"
+	schedule_interval = 50
+	cpu_threshold = 20
```
Comment (stuicey) [inline code/controllers/Processes/overmap.dm:30]: CPU threshold doesn't exist anymore

File: code/controllers/verbs.dm [hunk 2]
```diff
@@ -56,7 +56,7 @@
 		usr.client.debug_variables(antag)
 		message_admins("Admin [key_name_admin(usr)] is debugging the [antag.role_text] template.")
 */
-/client/proc/debug_controller(controller in list("Master","Ticker","Ticker Process","Air","Air Process","Machine Process","Lighting Process","Obj Process","Mob Process", "Jobs","Sun","Radio","Supply","Shuttles","Emergency Shuttle","Configuration","pAI", "Cameras", "Transfer Controller", "Gas Data","Event","Faction","Plants","Alarm","Nano","Templates","Template Config"))
+/client/proc/debug_controller(controller in list("Master","Ticker","Ticker Process","Air","Air Process","Machine Process","Lighting Process","Obj Process", "Overmap","Mob Process", "Jobs","Sun","Radio","Supply","Shuttles","Emergency Shuttle","Configuration","pAI", "Cameras", "Transfer Controller", "Gas Data","Event","Faction","Plants","Alarm","Nano","Templates","Template Config"))
```
Comment (stuicey) [inline code/controllers/verbs.dm:59]: Reworked this proc you can revert all changes to it

File: code/controllers/Processes/overmap.dm (hunk 1 above)
Comment (Kwask) [inline code/controllers/Processes/overmap.dm:30]: I'll let you deal with merging those changes from master :worried:

File: code/controllers/Processes/overmap.dm [hunk 3]
```diff
@@ -0,0 +1,196 @@
+#define MIN_ASTEROIDS 3
+#define MAX_ASTEROIDS 6
+
+/var/global/datum/controller/process/overmap/overmap
```
Comment (stuicey) [inline code/controllers/Processes/overmap.dm:4]: Don't need this anymore due to reworked debug variables

**Pull request merged**